docs/gitbook/supervised_learning/tutorial.md

```diff
@@ -0,0 +1,461 @@
+| category#book | 0.7337419390678406 |
+| category#electronics | 0.7443372011184692 |
+| category#entertainment | 5.039264578954317e-07 |
+| category#food | 1.415687620465178e-07 |
+| category#sports | -0.2697771489620209 |
+| gender#male | 0.7336684465408325 |
+| gender#female | 0.47442761063575745 |
+| gender#other | 5.039264578954317e-07 |
+| price | -110.62307739257812 |
+
+Notice that weight is learned for each possible value in a categorical feature, and for every single quantitative feature.
+
+Of course, you can optimize hyper-parameters to build more accurate prediction model. Check the output of the following query to see all available options, including learning rate, number of iterations and regularization parameters, and their default values:
+
+```sql
+select train_classifier(array(), 0, '-help');
+```
+
+### Step 3. Prediction
+
+Now, the table `classifier` has liner coefficients for given features, and we can predict unforeseen samples by computing a weighted sum of their features.
+
+How about the probability of purchase by a `male` customer who sees a `food` product priced at `120` on `Friday`? Which product is more likely to be purchased by the customer on `Friday`?
+
+To differentiate potential purchases, create a `unforeseen_samples` table with these unknown combinations of features:
+
+```sql
+create table if not exists unforeseen_samples as
+select 1 as id, array("gender#male", "category#food", "day of week#Friday", "price:120") as features
+union all
+select 2 as id, array("gender#male", "category#sports", "day of week#Friday", "price:1000") as features
+union all
+select 3 as id, array("gender#male", "category#electronics", "day of week#Friday", "price:540") as features
+;
+```
+
+Prediction for the feature vectors can be made by join operation between `unforeseen_samples` and `classifier` on each feature as:
+
+```sql
+with features_exploded as (
+  select
+    id,
+    -- split feature string into its name and value
+    -- to join with a model table
+    extract_feature(fv) as feature,
+    extract_weight(fv) as value
+  from unforeseen_samples t1 LATERAL VIEW explode(features) t2 as fv
+)
+select
+  t1.id,
+  sigmoid( sum(p1.weight * t1.value) ) as probability
+from
+  features_exploded t1
+  LEFT OUTER JOIN classifier p1 ON (t1.feature = p1.feature)
+group by
+  t1.id
+;
+```
+
+Output for single sample can be:
+
+|id| probability|
+|---:|---:|
+| 1| 1.0261879540562902e-10|
+
+### Evaluation
+
+If you have test samples for evaluation, use Hivemall's [evaluation UDFs](../eval/binary_classification_measures.html) to measure the accuracy of prediction.
+
+For instance, prediction accuracy over the `training` samples can be measured as:
+
+```sql
+with features_exploded as (
+  select
+    id,
+    extract_feature(fv) as feature,
+    extract_weight(fv) as value
+  from training t1 LATERAL VIEW explode(features) t2 as fv
+),
+predictions as (
+  select
+    t1.id,
+    sigmoid( sum(p1.weight * t1.value) ) as probability
```

Comment:
It might be better to note that sigmoid can only applied for logistic loss.
Cannot get a probability for other loss functions for a classification.

https://www.coursera.org/lecture/machine-learning/decision-boundary-WuL1H

Reply:
agreed. I will add the note for sigmoid at [the first appearance](https://github.com/apache/incubator-hivemall/pull/158/files#diff-00bf5d6dc412242452a06beca32c2f08R241) in this doc.

Reply:
Added 10cf627